icicle/include/vec_ops/vec_ops.cuh

```diff
@@ -113,6 +113,27 @@ namespace vec_ops {
     device_context::DeviceContext& ctx,
     bool on_device,
     bool is_async);
+
+  struct BitReverseConfig {
```

Comment:
(1) looks like the vecOpsConfig so consider reusing it instead of adding a new struct.
(2) for inplace, input==output but the config may tell you that one is on device but the other is not. How do handle that?

Reply:
(2) I use only `is_output_on_device` for inplace because `input` isn't used in this scenario.